### DISCUSSION TITLE
bug两条

### DISCUSSION THREAD

**Discussion by VansamaZWT** [liusy58/NJU_AdvancedProgramming_SP22#183] [Q&A]
遇到了我无法解决的两个bug，我简单描述一下下面的代码，主要作用就是实现遍历存放所有敌人的链表(all_enemies)（在下面的代码中为了便于测试我只检查了链表头部），通过检查敌人的所在格子指针(position)是否与重点指针相等来判断是否将成员数据lose设为true。我不理解上下代码只相差一个this，为何上方可运行下方则不可，恳请大佬帮助
![ESY(M7UL0T%Y9JEE1$4 8`X](https://user-images.githubusercontent.com/100075314/165293588-0fdefa68-ee14-4ed4-bac5-53a738262608.png)

**Comment by VansamaZWT** [comment]
bug之二是为何使用成员函数无法运行，将其代码copy出来就可以运行，代码如下
![R3JN57O`_O D716KJ@P3QD0](https://user-images.githubusercontent.com/100075314/165293792-74d223d6-4d01-47f1-bdda-42f7743aafb6.png)

**Comment by RicoloveFeng** [comment]
请补充一些信息：
1. 可运行和不可运行有什么区别？报错是什么？不可运行是闪退，还是编译错误，还是运行时报错，还是功能不符合预期？
2. 你是否重定义了lose使其成为了一个局部变量？
3. 你有没有观察这两条语句执行之后，lose在内存里面变成了什么值？是否排除了逻辑判断本身出问题而导致功能出问题？

**Reply by VansamaZWT** [reply]
谢谢您的指导，2.是错误的根源。我在初始化时写了如下语句“bool lose=false;”，恐怕这使得lose被重定义为了局部变量。
